src/definitions.jl

```diff
@@ -0,0 +1,74 @@
+
+using LightGraphs, MetaGraphs
+
+import JSON
+
+# TODO: Create functions for each
+# Cause - is concept label, Effect - definition label
+# within a RelationMention with labels "Definition, Entity"
+
+definitions_output_file = "/Users/scott/Documents/JuliaProjects/SemanticModels.jl/test/definitions.jl/automates_output_1sentence_definition.json"
+def_output_dir = "/Users/scott/Documents/JuliaProjects/SemanticModels.jl/test/definitions.jl"
+
+
+function info_extract_from_automates(output_dir::String)
+    file_list = readdir(output_dir)
+    graph = DiGraph()
+    changes_recorded = []
+    v_count = 0
+    for def_file in file_list
+        json_res = JSON.parsefile(string(def_output_dir, "/",def_file))
+        # json_res = JSON.parsefile(definitions_output_file)
+        mentions = json_res["mentions"]
+        # println("JSON Mentions is:\n")
+        # println(mentions)
+        # println("End mentions\n")
+        for val in mentions
+            if haskey(val,"type") && val["type"] == "RelationMention"
+                JSON.print(val, 4)
+                v_count += 1
+                cause = val["arguments"]["cause"][1]["text"]
+                effect = val["arguments"]["effect"][1]["text"]
+                println(string("Cause is: ", cause, "\n"))
+                println(string("Effect is: ", effect, "\n"))
+                println("Add new vertex for Concept to Graph")
+                add_vertex!(graph)
```

Comment:
Don't you need to check that the concept doesn't already exist in the graph?

Reply:
agree with @jpfairbanks that we'll need to check if v \in V already; as a first pass, we'll have to assign hashed vertex ids upon ingestion (for now, I think we'll just hash the text) and then check the hash against existing vertex ids. We should probably think about how to extend this to a fuzzy matching-type approach as the graph grows though (e.g., requiring an exact hash match amounts to an exact string match, unless we are hashing multiple metadata; semantic equivalence is ultimately what we're after here).

Reply:
Yeah we can just use a Dict{String, Int}. MetaGraphs has support for this already. If we convert it to a dataframe with columns `src,dst,prop1, prop2...`, then we can ingest it easily into MG.

Reply:
Yeah; the example nb I made earlier does this, but I did not handle the id generation in that notebook. Will be an easy fix.